foundry/roll.d.ts

```diff
@@ -191,16 +171,12 @@ declare class Roll<D = object> {
    */
   roll(): this;
 
-  /* -------------------------------------------- */
-
   /**
    * Create a new Roll object using the original provided formula and data
    * Each roll is immutable, so this method returns a new Roll instance using the same data.
    * @returns A new Roll object, rolled using the same formula and data
    */
-  reroll(): Roll<D>;
-
-  /* -------------------------------------------- */
+  reroll(): this;
```

Comment:
See `clone`

Reply:
Yes, but they use `new this.constructor(...)`, so if you derive from `Roll` and call `clone` on that derived class, you'll get an instance of the derived class.